superset/daos/dataset.py

```diff
@@ -184,15 +185,24 @@ def update(
         """
 
         if item and attributes:
+            force_update: bool = False
             if "columns" in attributes:
                 cls.update_columns(
                     item,
                     attributes.pop("columns"),
                     override_columns=bool(attributes.get("override_columns")),
                 )
+                force_update = True
 
             if "metrics" in attributes:
                 cls.update_metrics(item, attributes.pop("metrics"))
+                force_update = True
+
+            # columns and metrics are stored in separate tables so if they change we
+            # force an update to the dataset too to ensure the last_updated_timestamp
+            # is updated (it's possible no other dataset attributes are changed).
+            if force_update:
```

Comment:
@Vitor-Avila Why not be explicit about this operation and update the `changed_on` field on the dataset? It would avoid the whole comments and the cognitive load to understand that `changed_on` will be automatically updated once the record is changed.

Reply:
@michael-s-molina are you suggesting we always do it? 

Technically we could. I opted for this flow because we only need to worry with this in case columns and/or metrics were updated (and no direct dataset attribute was). I don't think there would be a side-effect of always doing, it's just one extra query.

Reply:
I think he's suggesting that instead of updating the last modified column as a side effect like you do on line 205, do something that makes the intent clear, something like:

```python
dataset = db.session.query(SqlaTable).get(item.id)
dataset.changed_on = XXX
```

Reply:
alright, I added `changed_on` to `attributes` so it's processed as any other column.